tensorflow/python/ops/array_ops.py

```diff
@@ -4487,6 +4487,21 @@ def where_v2(condition, x=None, y=None, name=None):
   <tf.Tensor: shape=(4,), dtype=int32, numpy=array([100, 100, 100, 100],
   dtype=int32)>
 
+  Note that if the gradient of either branch of the tf.where generates
+  a NaN, then the gradient of the entire tf.where will be NaN.
+  A workaround is to use an inner tf.where to ensure the function has
+  no asymptote, and to avoid computing a value whose gradient is NaN by
+  replacing dangerous inputs with safe inputs.
+  
+  Instead of this
+  
+  >>> y = float(-1)
+  >>> tf.where(y > 0, tf.sqrt(y), y)
```

Comment:
Can you also add the output please? This would make this a real test. Same on the sequence below.

Also, let's make `y` a `tf.constant` tensor, fixing its type to (let's say) `tf.float32`

Reply:
We now have 2 examples saying the same thing. Why not just one example?

```python

That is, instead of this:

>>> y = tf.constant(01, dtype=tf.float32)
>>> tf.where(y > 0, tf.sqrt(y), y)
INSERT OUTPUT HERE

Use this:

>>> tf.where(y > 0, tf.sqrt...
<tf.Tensor...
```